### ISSUE TITLE
Policy Execution Failing Silently with No Reason

### ISSUE THREAD

**Issue by bbernays** [cloudquery/cloudquery#304]
### Describe the Bug
If you try and run a policy and there is no provider downloaded it will fail silently

``` bash
./cloudquery policy run -v
10:58PM INF Starting policies run...
10:58PM INF Loading policy args={"Name":"public-ips","Source":"./policy.hcl","SubPath":"","Type":"local","Version":"latest"}
10:58PM DBG Policy %s finished with error: %v public-ips="provider aws doesn't exist"
10:58PM INF Loading policy args={"Name":"public-ips2","Source":"policy.hcl","SubPath":"","Type":"local","Version":"latest"}
10:58PM DBG Policy %s finished with error: %v public-ips2="provider aws doesn't exist"
10:58PM INF Loading policy args={"Name":"public-ips3","Source":"../policy.hcl","SubPath":"","Type":"local","Version":"latest"}
10:58PM DBG Policy %s finished with error: %v public-ips3="provider aws doesn't exist"
10:58PM INF Finished policies run...


./cloudquery policy run
10:58PM INF Starting policies run...
10:58PM INF Loading policy args={"Name":"public-ips","Source":"./policy.hcl","SubPath":"","Type":"local","Version":"latest"}
10:58PM INF Loading policy args={"Name":"public-ips2","Source":"policy.hcl","SubPath":"","Type":"local","Version":"latest"}
10:58PM INF Loading policy args={"Name":"public-ips3","Source":"../policy.hcl","SubPath":"","Type":"local","Version":"latest"}
10:58PM INF Finished policies run...

```

### Expected Behavior

I would expect that either cloudquery will download the provider (as it is fully defined in the configl.hcl file) or at least return an error rather than saying that the policies finished successfully

### CloudQuery Version

```
Version: 0.16.1
Commit: 8f8550a92685eab9b49ea99efe1e185ee15b73c6
Date: 2021-11-27 22:58:05.108961969 +0000 UTC m=+0.011160834
```
### Steps to Reproduce

``` bash
curl -L https://github.com/cloudquery/cloudquery/releases/latest/download/cloudquery_Linux_arm64 -o cloudquerycloudquery fetch
chmod +x ./cloudquery
./cloudquery version
./cloudquery policy run -v
./cloudquery policy run
```

**Comment by roneli** [comment]
@bbernays does this reproduce in v0.16.2? added some error prints there, might need to add more.

**Comment by bbernays** [comment]
I don't think it does fix it. You still have to use the `-v` option to get any information about why it is not running:
``` bash
agent_1  | [Container] 2021/12/01 12:40:36 Running command ./cloudquery version
agent_1  | 12:40PM INF logging configured consoleLog=false fileLogging=true fileName=cloudquery.log jsonLogOutput=false logDirectory=. maxAgeInDays=3 maxBackups=3 maxSizeMB=30 verbose=false
agent_1  | Version: 0.16.2
agent_1  | Commit: b84745a3e34df4169321aa7365d5fdbf2bc3d32b
agent_1  | Date: 2021-12-01 12:40:36.18000922 +0000 UTC m=+0.013967042
agent_1  | 
agent_1  | [Container] 2021/12/01 12:40:36 Running command ./cloudquery policy run
agent_1  | 12:40PM INF logging configured consoleLog=false fileLogging=true fileName=cloudquery.log jsonLogOutput=false logDirectory=. maxAgeInDays=3 maxBackups=3 maxSizeMB=30 verbose=false
agent_1  | 12:40PM WRN failed to read plugin directory, no existing plugins loaded directory=./.cq/providers
agent_1  | 12:40PM INF Policies to run policies=[{"Name":"remote_policy_1","Source":"https://github.com/cloudquery-policies/aws-cis-1.2.0.git","SubPath":"","Type":"remote","Version":"v0.0.5"}]
agent_1  | 12:40PM INF Starting policies run...
agent_1  | 12:40PM INF Loading policy args={"Name":"remote_policy_1","Source":"https://github.com/cloudquery-policies/aws-cis-1.2.0.git","SubPath":"","Type":"remote","Version":"v0.0.5"}
agent_1  | 12:40PM INF Finished policies run...
agent_1  | 
agent_1  | [Container] 2021/12/01 12:40:36 Running command ./cloudquery policy run -v
agent_1  | 12:40PM INF logging configured consoleLog=false fileLogging=true fileName=cloudquery.log jsonLogOutput=false logDirectory=. maxAgeInDays=3 maxBackups=3 maxSizeMB=30 verbose=true
agent_1  | 12:40PM WRN failed to read plugin directory, no existing plugins loaded directory=./.cq/providers
agent_1  | 12:40PM DBG Received params: args: %v, policyName: %s, outputDir: %s, stopOnFailure: %v, skipVersioning: %v, failOnViolation: %v, noResults: %v =false []= false=null
agent_1  | 12:40PM INF Policies to run policies=[{"Name":"remote_policy_1","Source":"https://github.com/cloudquery-policies/aws-cis-1.2.0.git","SubPath":"","Type":"remote","Version":"v0.0.5"}]
agent_1  | 12:40PM INF Starting policies run...
agent_1  | 12:40PM INF Loading policy args={"Name":"remote_policy_1","Source":"https://github.com/cloudquery-policies/aws-cis-1.2.0.git","SubPath":"","Type":"remote","Version":"v0.0.5"}
agent_1  | 12:40PM DBG Policy execution finished err="provider aws doesn't exist" name=remote_policy_1
agent_1  | 12:40PM INF Finished policies run...
agent_1  | 
agent_1  | [Container] 2021/12/01 12:40:36 Phase complete: BUILD State: SUCCEEDED
agent_1  | [Container] 2021/12/01 12:40:36 Phase context status code:  Message: 
agent_1  | [Container] 2021/12/01 12:40:36 Entering phase POST_BUILD
agent_1  | [Container] 2021/12/01 12:40:36 Phase complete: POST_BUILD State: SUCCEEDED
agent_1  | [Container] 2021/12/01 12:40:36 Phase context status code:  Message: 
```

**Comment by bbernays** [comment]
Fixed in`0.17.0`.... Logs now indicate to user the underlying issue
